src/middleware/navigation/has.trust.middleware.ts

```diff
@@ -28,6 +29,54 @@ const hasTrustWithId = (req: Request, res: Response, next: NextFunction, url: st
   }
 };
 
+const hasTrusteeWithId = (req: Request, res: Response, next: NextFunction, url: string): void => {
+  try {
+    const trustId = req.params[ROUTE_PARAM_TRUST_ID];
+    const trusteeId = req.params[ROUTE_PARAM_TRUSTEE_ID];
+    const trusteeType = req.params[ROUTE_PARAM_TRUSTEE_TYPE];
+    const appData: ApplicationData = getApplicationData(req.session);
+
+    const isTrustPresent = appData[TrustKey]?.some(
+      (trust) => trust.trust_id === trustId,
+    );
+
+    if (!isTrustPresent) {
+      logger.infoRequest(req, NavigationErrorMessage);
+
+      return res.redirect(url);
+    }
+
+    let hasTrusteeForChange = false;
+
+    const HISTORICAL = appData[TrustKey]?.find(t => t.trust_id === trustId)?.HISTORICAL_BO ?? [];
+    const INDIVIDUAL = appData[TrustKey]?.find(t => t.trust_id === trustId)?.INDIVIDUALS ?? [];
+    const CORPORATE = appData[TrustKey]?.find(t => t.trust_id === trustId)?.CORPORATES ?? [];
+
+    switch (trusteeType as TrusteeType){
+        case TrusteeType.HISTORICAL:
+          hasTrusteeForChange = HISTORICAL.some(t => t.id === trusteeId);
+          break;
+        case TrusteeType.INDIVIDUAL:
+          hasTrusteeForChange = INDIVIDUAL.some(t => t.id === trusteeId);
+          break;
+        case TrusteeType.LEGAL_ENTITY:
+          hasTrusteeForChange = CORPORATE.some(t => t.id === trusteeId);
+          break;
+        default:
+          break;
+    }
+
+    if (!hasTrusteeForChange && trusteeId) {
```

Comment:
seems a bit late to worry about trusteeId being present? Maybe handle that earlier

Reply:
Fixed

Reply:
It was to handle the trust change link too but moved to a different approach